### PR TITLE
fix: extensions list order and copy update

### PR DESCRIPTION
**Pull request by joanna-pagepro**
## Description

Extensions list order

## Testing

* Go to the Extensions page.
* Note the order and grouping of extensions.
* Install an extension.
* Note the order of extensions.

## Diffs

**New stuff** ✨

* 

**Changes** 🏗

* `supportedExtensionsConfig` split into `paymentSupportedExtensionsConfig` and `decisionMethodSupportedExtensionsConfig`
* `StakedExpenditureSettings` component updated to use `initializationParams` array of the extension

**Deletions** ⚰️

* 

## TODO

- [ ] Add TODOs

Resolves https://github.com/JoinColony/colonyCDapp/issues/2928

### CODE REVIEW

**Review by rumzledz — APPROVED** [review]
Awesome stuff @joanna-pagepro , I can confirm that the order is correct based on #2928 

Screenshot 2024-08-23 at 01 10 00

File: src/components/frame/Extensions/pages/Extensions/ExtensionsListingPage/utils.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    );
+
+    if (availableExtensionInfo) {
+      acc.push(availableExtensionInfo);
+    } else if (installedExtensionInfo) {
+      acc.push(installedExtensionInfo);
+    }
+
+    return acc;
+  }, []);
```
Comment (mmioana) [inline src/components/frame/Extensions/pages/Extensions/ExtensionsListingPage/utils.ts:34]: I would propose refactoring this into
```suggestion
  items
    .map(item => item?.extensionId)
    .filter(notMaybe)
    .reduce<AnyExtensionData[]>((acc, extensionId) => {

    let extensionData = availableExtensionsData.find(
      (extension) => extension.extensionId === extensionId,
    );
    
    if (!extensionData) {
      extensionData =  installedExtensionsData.find(
        (extension) => extension.extensionId === extensionId,
      );
    }

    if (extensionData) {
      acc.push(extensionData);
    } 
    
    return acc;
  }, []);
```

File: src/components/frame/Extensions/pages/Extensions/ExtensionsListingPage/ExtensionsListingPage.tsx [hunk 2]
```diff
@@ -19,34 +24,44 @@ const MSG = defineMessages({
...
-      return acc;
-    }, {});
+  const categorizedExtensions = [
+    {
+      category: formatText(MSG.decisionMethods),
+      extensions: getExtensionData(
+        decisionMethodSupportedExtensionsConfig,
+        availableExtensionsData,
+        installedExtensionsData,
+      ).sort((a, b) => formatText(a.name).localeCompare(formatText(b.name))),
```
Comment (mmioana) [inline src/components/frame/Extensions/pages/Extensions/ExtensionsListingPage/ExtensionsListingPage.tsx:49]: Could we please extract the sorting function into an utility function
```
const sortByName = (extensionA: AnyExtensionData, extensionB: AnyExtensionData) => 
  formatText(extensionA.name).localeCompare(formatText(extensionB.name)))
```
and then use it here as `.sort(sortByName)`

**Review by mmioana — CHANGES_REQUESTED** [review]
Really nice work on this one @joanna-pagepro ❤️ 

I left some refactoring comments, if you could please be so kind to tackle them 🙏

**Comment by joanna-pagepro** [comment]
@mmioana done

**Review by mmioana — APPROVED** [review]
Thank you @joanna-pagepro for your changes 🙏

**Pull request merged**